### PR TITLE
ACCA_IfcTunnel_Sprint_2_1

### PR DESCRIPTION
**Pull request by michelangelo-acca**
## File

The IFC files produced from ACCA software for the Sprint 2_1.

It contains:

- Project
- UM
- Context
- A Group with a Referent
- Project
- Spatial structures
- Alignment (Semantic & Geometric)
- Referent

## Covered Concept Templates - Usages Included

The covered concept templates are:

1. SpatialComposition
2. SpatialDecomposition
3. LinearComposition
4. AlignmentLayout
5. ObjectNesting
6. GroupAssignment

### CODE REVIEW

**Review by larswik — COMMENTED** [review]
Regarding the covered templates, I think that the one failing is "Each IfcTunnel may **only** be decomposed by IfcTunnel or IfcTunnelPart.". The IfcTunnel currently has IfcSpatialZone instances as well.

**Comment by michelangelo-acca** [comment]
> Regarding the covered templates, I think that the one failing is "Each IfcTunnel may **only** be decomposed by IfcTunnel or IfcTunnelPart.". The IfcTunnel currently has IfcSpatialZone instances as well.

Thanks for the feedback, will try to update the file asap.

**Comment by pjanck** [comment]
@michelangelo-acca Is an update of the file to support usages from Sprints 1.x planned? Or should we already review the file?

**Comment by michelangelo-acca** [comment]
> @michelangelo-acca Is an update of the file to support usages from Sprints 1.x planned? Or should we already review the file?

All concepts are supported except the ones from Sprint 1.3. We will try to implement them later on and update the files but for now they are unsupported!

**Comment by michelangelo-acca** [comment]
The model has been updated...can you spot why the 'Alignment representation in Sprint 2.2: horizontal curve' fails?

**Comment by SergejMuhic** [comment]
> The model has been updated...can you spot why the 'Alignment representation in Sprint 2.2: horizontal curve' fails?

I think you are missing this:
image

**Comment by michelangelo-acca** [comment]
> > The model has been updated...can you spot why the 'Alignment representation in Sprint 2.2: horizontal curve' fails?
> 
> I think you are missing this: image

Fixed, thanks!

**Comment by SergejMuhic** [comment]
> Fixed, thanks!

looks good now.

**Review by larswik — APPROVED** [review]
I think that the file lftm. It is only the sprint 1.3 concepts that are failing.

**Review by pjanck — APPROVED** [review]
lftm

I wonder however: given the coordinates in the file, where exactly should the geometries reside on the Earth? It seems that these are very close to the origin of 'EPSG:3065'?

**Comment by michelangelo-acca** [comment]
> lftm
> 
> I wonder however: given the coordinates in the file, where exactly should the geometries reside on the Earth? It seems that these are very close to the origin of 'EPSG:3065'?

Fixed, now with MapConversion information!

**Comment by michelangelo-acca** [comment]
Added ACCA_IfcTunnel_Sprint_2_2.ifc for the Sprint_2_2, which is the same as the Sprint_2_1. 

Not sure how to split it in two separate pull requests...

File: files/ACCA/ACCA_IfcTunnel_Sprint_2_1.ifc [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+#349 = IFCRELAGGREGATES('04uEZHi9TC_ucBD5yxc4vZ', #1, 'Spatial Structure Element Container', 'Spatial Structure Element to Facilities', #97, (#102));
+#350 = IFCRELAGGREGATES('1xuXq3$E9Esutjr6L7iWh$', #1, 'Project Container', 'Project to Spatial Element', #7, (#97));
+#351 = IFCGROUP('2shLc4pvj8Bwg0qya49XO_', #1, 'Referents', $, $);
+#352 = IFCRELASSIGNSTOGROUP('3Eyv$LEmbC6wAACZz7PD2Y', #1, 'Contained in Group', 'Contained in Group', (#235), $, #351);
+#353 = IFCRELDECLARES('1cyZjVSfzFlfBHLdqcJDtm', #1, 'Project Declaration Relationship', $, #7, (#351));
+#354 = IFCRELNESTS('2eswLs9SbFxfimlFUPgr7$', #1, 'Nests', $, #107, (#235));
+#355 = IFCRELPOSITIONS('1iY89mEMf2n8VN2VQvgiL8', #1, 'Positions', $, #107, (#235));
+#356 = IFCRELAGGREGATES('3ovfhxlK96UwZs$Z$6PuBj', #1, 'Project aggregation', $, #7, (#107));
+#357 = IFCMAPCONVERSION(#13, #358, 235916.629974235, 5192108.10113525, 937., 5.E-1, 8.66025403784439E-1, 1.);
+#358 = IFCPROJECTEDCRS('EPSG:3065', 'Istituto Geografico Militare 1995 (IGM95)', 'EPSG:3065', 'EPSG:5214', 'UTM', '33N', $);
```
Comment (pjanck) [inline files/ACCA/ACCA_IfcTunnel_Sprint_2_1.ifc:365]: This seems plausible! Thank you for correcting.

**Pull request merged**